### PR TITLE
fix warnings on linux

### PR DESCRIPTION
**Pull request by shaomeng**
This PR has the following changes:
1. It removes compiler flags that suppress certain warnings.
2. It adds global flags "-Wall -Wno-sign-compare" that reports warnings.
3. It fixes most VAPOR warnings on linux using GCC.

What is left:
1. A few warnings that need attention from Scott. I'll submit issues once this PR is approved.
2. Clang has additional warnings. We'll examines them in the next steps.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -35,6 +35,13 @@ elseif (WIN32)
 	add_definitions (-DWIN32)
 endif()
 
+# compiler warning flags
+if (NOT WIN32)
+  SET( COMPILE_FLAGS "-Wall -Wno-sign-compare")
+  SET( CMAKE_CXX_FLAGS  "${COMPILE_FLAGS}" )
+  SET( CMAKE_C_FLAGS  "${COMPILE_FLAGS}" )
```
Comment (StasJ) [inline CMakeLists.txt]: We don't have any c code

File: apps/vaporgui/main.cpp [hunk 2]
```diff
@@ -166,7 +166,6 @@ if (getenv("VAPOR_DEBUG"))
     fontFile = fontFile + "//arimo.ttf";
 
 	QFontDatabase fdb; 
-	int result = fdb.addApplicationFont(QString::fromStdString(fontFile));
```
Comment (StasJ) [inline apps/vaporgui/main.cpp]: No

File: include/vapor/UnstructuredGridCoordless.h [hunk 3]
```diff
@@ -137,9 +137,6 @@ void GetIndices(
   virtual const void *address() const {return this; };
 
   virtual bool equal(const void* rhs) const {
-	const ConstCoordItrUCoordless *itrptr = 
-		static_cast<const ConstCoordItrUCoordless *> (rhs);
-
```
Comment (StasJ) [inline include/vapor/UnstructuredGridCoordless.h]: What?

File: lib/render/ImageRenderer.cpp [hunk 4]
```diff
@@ -201,8 +200,6 @@ int ImageRenderer::GetMesh( DataMgr *dataMgr,
 		// Get the width and height of the image texture. These
 		// will be used to set the width and height of the mesh.
 		//
-		unsigned char *texture = (unsigned char *) _getTexture(dataMgr);
```
Comment (StasJ) [inline lib/render/ImageRenderer.cpp]: This will break things.

File: CMakeLists.txt (hunk 1 above)
Comment (shaomeng) [reply CMakeLists.txt]: hmm, how about this one:
https://github.com/NCAR/VAPOR/blob/53b1fa78b25c1f84e1c1ca8680c8b849de71e79f/lib/vdc/kdtree.c

Comment (StasJ) [reply CMakeLists.txt]: Fair point

File: apps/vaporgui/main.cpp (hunk 2 above)
Comment (shaomeng) [reply apps/vaporgui/main.cpp]: good call! 

Fixed in commit 104b214 .

File: include/vapor/UnstructuredGridCoordless.h (hunk 3 above)
Comment (shaomeng) [reply include/vapor/UnstructuredGridCoordless.h]: Agreed. 

That fix will go into a different PR though.

File: lib/render/ImageRenderer.cpp (hunk 4 above)
Comment (shaomeng) [reply lib/render/ImageRenderer.cpp]: More specifically?

Comment (shaomeng) [reply lib/render/ImageRenderer.cpp]: fixed in commit 4e9502a

File: apps/vaporgui/ErrorReporter.cpp [hunk 5]
```diff
@@ -197,7 +197,7 @@ string ErrorReporter::GetSystemInformation()
 	ret += "OS: " + string(info.sysname) + " " + string(info.release) + " " + string(info.version) + "\n";
 	ret += "Distro:\n";
 	char buffer[128];
-    	FILE *pipe = popen("lsb_release", "r"), pclose;
+    	FILE *pipe = popen("lsb_release", "r");
```
Comment (clyne) [inline apps/vaporgui/ErrorReporter.cpp]: pipe is not being closed

File: lib/wasp/WASP.cpp [hunk 6]
```diff
@@ -455,12 +456,15 @@ vector <size_t> vdiff(vector <size_t> a, vector <size_t> b) {
 	for (int i=0; i<a.size(); i++) c[i] = a[i] - b[i];
 	return(c);
 }
+#endif
 
 // Determine POD type 
 //
+#ifdef VAPOR3_0_0_ALPHA
```
Comment (StasJ) [inline lib/wasp/WASP.cpp]: If this is commented out a call to NetCDFType((float)3.0f) will be promoted to NetCDFType(double) without warning

Comment (shaomeng) [reply lib/wasp/WASP.cpp]: fixed in commit 4d331b0

File: apps/vaporgui/ErrorReporter.cpp (hunk 5 above)
Comment (shaomeng) [reply apps/vaporgui/ErrorReporter.cpp]: That's a bug of the code. I intended to only address warnings in this PR, unless you think I should fix bugs along the way.

**Comment by shaomeng** [comment]
Don't merge yet, more fixes coming...

File: apps/vaporgui/ErrorReporter.cpp (hunk 5 above)
Comment (shaomeng) [reply apps/vaporgui/ErrorReporter.cpp]: since this bug is so easy to fix, I fixed via commit 499f0ba

**Pull request merged**